scammer-label-propagation/src/main.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+import numpy as np
+import pandas as pd
+import torch
+from forta_agent import get_json_rpc_url
+from hexbytes import HexBytes
+from web3 import Web3
+
+from src.constants import (ATTACKER_CONFIDENCE, MIN_FOLDS_ATTACKER, N_FOLDS,
+                           PREDICTED_ATTACKER_CONFIDENCE, VICTIM_SAMPLING)
+from src.model.aux import cross_entropy_masked
+from src.model.model import ModelAttention
+from src.model.train import prepare_graph_and_train
+from src.preprocessing.get_data import (collect_data_parallel_parts,
+                                        download_labels_graphql,
+                                        get_automatic_labels)
+from src.preprocessing.process_data import prepare_data
+
+logger = logging.getLogger(__name__)
+
+
+def run_all(central_node):
+    """
+    Run the whole pipeline for a given central node. Returns a df with the new addresses classified as attackers
+    :param central_node: str with the node to build the graph around and analyze
+    :return: pd.DataFrame with the new attackers and the probabilities
+    """
+    central_node = central_node.lower()
+    logger.info(f"{central_node}:\tStart processing")
+
+    model_type = ModelAttention
+    loss_function = cross_entropy_masked
+
+    data = collect_data_parallel_parts(central_node)
+    all_nodes_dict, node_feature, transactions_overview, edge_indexes, edge_features = prepare_data(data)
+    labels_df = download_labels_graphql(all_nodes_dict, central_node)
+    np.random.seed(1993)
```

Comment:
might be good to add the random seed as a constant in the constants file

Reply:
done